k8s/migration/internal/controller/migrationplan_controller.go

```diff
@@ -274,6 +275,10 @@ func (r *MigrationPlanReconciler) CreatePod(ctx context.Context,
 	vmname := strings.ReplaceAll(strings.ReplaceAll(vm, " ", "-"), "_", "-")
```

Comment:
Let's ensure the VM name is RFC1123 compliant. Eg. if vm comes in as `_name` this will fail to create the POD due to the name beginning with `-`.